### ISSUE TITLE
AMUSE on Notebook Server (Dockerized Containers)

### ISSUE THREAD

**Issue by JPGlaser** [amusecode/amuse#778]
**Describe the bug**
When attempting to use AMUSE on a Dockerized Notebook instance on our JupyterHub server, processes that call for `worker` codes cause MPI failures. See: https://github.com/open-mpi/ompi/issues/9113

**To Reproduce**
Install AMUSE via Source or pip onto `https://hub.docker.com/r/nanograv/nanopulse`. Run any example code which uses a `worker` subprocesses.

**Expected behavior**
Workers should be allocated to one of the 20 CPUs available as per normal operation. This currently occurs with basic HelloWorld programs that call on the same libraries within the DockerContainer.

**Logs**
See: https://github.com/open-mpi/ompi/issues/9113

**Environment (please complete the following information):**
 - OS and version: Ubuntu 20 LTS via Docker
 - Environment: nanograv/nanopulse Docker (see https://hub.docker.com/r/nanograv/nanopulse for full environment variables)

**Comment by JPGlaser** [comment]
So I think the key here is this:
```
(base) jovyan@c4cd802d7fc0:~/work/shared/amuse$ ./amuse.sh examples/simple/massloss.py
--------------------------------------------------------------------------
At least one pair of MPI processes are unable to reach each other for
MPI communications.  This means that no Open MPI device has indicated
that it can be used to communicate between these processes.  This is
an error; Open MPI requires that all MPI processes be able to reach
each other.  This error can sometimes be the result of forgetting to
specify the "self" BTL.

  Process 1 ([[11747,2],0]) is on host: c4cd802d7fc0
  Process 2 ([[11747,1],0]) is on host: unknown!
  BTLs attempted: self tcp

Your MPI job is now going to abort; sorry.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  ompi_dpm_dyn_init() failed
  --> Returned "Unreachable" (-12) instead of "Success" (0)
--------------------------------------------------------------------------
```
Specifically, it seems that it is trying to place `Process 2 ([[11747,1],0])` (aka: the worker code), but can't find a place to put it. I wonder if its trying to place it on another host instead of self because it thinks its using all of self (despite there being 30+ cores on that 2U machine). @ipelupessy , have you seen something like this before with OpenMPI?

~ Joe G.

**Comment by spzwart** [comment]
Dear Joe et al.

I forward this message also to David Jansen, who is willing to look into
the docker/AMUSE issue.

Cheers,
Simon

**Comment by JPGlaser** [comment]
This has been solved by updating OpenMPI to version 4.1+. This is likely due to the fact that pip installs of mpi4py 3+ don't actually check themselves against the version of OpenMPI to roll back to a compatible subversion.

~ Joe G.
